core/src/main/scala/cats/syntax/vector.scala

```diff
@@ -1,11 +1,141 @@
 package cats.syntax
 
 import cats.data.NonEmptyVector
+import cats.{Applicative, Functor, Order, Traverse}
+
+import scala.collection.immutable.SortedMap
 
 trait VectorSyntax {
   implicit final def catsSyntaxVectors[A](va: Vector[A]): VectorOps[A] = new VectorOps(va)
 }
 
 final class VectorOps[A](private val va: Vector[A]) extends AnyVal {
+
+  /**
+   * Returns an Option of NonEmptyVector from a Vector
+   *
+   * Example:
+   * {{{
+   * scala> import cats.data.NonEmptyVector
+   * scala> import cats.implicits._
+   *
+   * scala> val result1: Vector[Int] = Vector(1, 2)
+   * scala> result1.toNev
+   * res0: Option[NonEmptyVector[Int]] = Some(NonEmptyVector(1, 2))
+   *
+   * scala> val result2: Vector[Int] = Vector.empty[Int]
+   * scala> result2.toNev
+   * res1: Option[NonEmptyVector[Int]] = None
+   * }}}
+   */
   def toNev: Option[NonEmptyVector[A]] = NonEmptyVector.fromVector(va)
+
+  /**
+   * Concatenates this `Vector` with a `NonEmptyVector` producing a new `NonEmptyVector`.
+   *
+   * Example:
+   * {{{
+   * scala> import cats.data.NonEmptyVector
+   * scala> import cats.implicits._
+   *
+   * scala> Vector(1, 2, 3).concatNev(NonEmptyVector.of(4, 5, 6))
+   * res0: NonEmptyVector[Int] = NonEmptyVector(1, 2, 3, 4, 5, 6)
+   * }}}
+   */
+  def concatNev[AA >: A](nev: NonEmptyVector[AA]): NonEmptyVector[AA] = nev.prependVector(va)
+
+  /**
+   * Groups elements inside this `Vector` according to the `Order` of the keys
+   * produced by the given mapping function.
+   *
+   * {{{
+   * scala> import cats.data.NonEmptyVector
+   * scala> import scala.collection.immutable.SortedMap
+   * scala> import cats.implicits._
+   *
+   * scala> val vector = Vector(12, -2, 3, -5)
+   *
+   * scala> val expectedResult = SortedMap(false -> NonEmptyVector.of(-2, -5), true -> NonEmptyVector.of(12, 3))
+   *
+   * scala> vector.groupByNev(_ >= 0) === expectedResult
+   * res0: Boolean = true
+   * }}}
+   */
+  def groupByNev[B](f: A => B)(implicit B: Order[B]): SortedMap[B, NonEmptyVector[A]] = {
+    implicit val ordering: Ordering[B] = B.toOrdering
+    toNev.fold(SortedMap.empty[B, NonEmptyVector[A]])(_.groupBy(f))
+  }
+
+  /**
+   * Groups elements inside this `Vector` according to the `Order` of the keys
+   * produced by the given mapping monadic function.
+   *
+   * {{{
+   * scala> import cats.data.NonEmptyVector
+   * scala> import scala.collection.immutable.SortedMap
+   * scala> import cats.implicits._
+   *
+   * scala> val vector = Vector(12, -2, 3, -5)
+   *
+   * scala> val expectedResult = Option(SortedMap(false -> NonEmptyVector.of(-2, -5), true -> NonEmptyVector.of(12, 3)))
+   *
+   * scala> vector.groupByNevA(num => Option(0).map(num >= _)) === expectedResult
+   * res0: Boolean = true
+   * }}}
+   */
+  def groupByNevA[F[_], B](
```

Comment:
I wonder if this should live here:
https://github.com/typelevel/cats/blob/main/core/src/main/scala/cats/NonEmptyTraverse.scala

```scala
def groupByA[G[_], B](fn: A => G[B])(implicit G: Apply[G], B: Order[B]): G[SortedMap[B, F[A]]]
```
and then you can call that here:
```scala
  toNev.fold(F.pure(SortedMap.empty[B, NonEmptyVector[A]])(_.groupByA(f))
```

Reply:
Well, these particular methods are just an attempt to maintain some degree of consistency and interoperability among all linear "non-empty" collections we have in `cats.data`. Because initially there was `ListOps#groupByNelA` only but no corresponding methods in `VectorOps` nor `SeqOps`. Which is a bit confusing.

Reply:
But I like the idea of generalization. I also have a feeling, that keeping stuffing various `*Ops` classes with almost identical methods is not the best way to deal with it. I like your suggestion in particular, but why `NonEmptyTraverse` ?

Seems, it should work with just a regular `Traverse` – shouldn't it?

I mean, when we do grouping on some collection by some predicate, all the groups we will get eventually are going to be non-empty regardless of whether the source collection was empty or not. I.e. we will either get a non-empty group or don't get it at all.

Reply:
We can use the way you suggested of course:
```
toNev.fold(F.pure(SortedMap.empty[B, NonEmptyVector[A]])(_.groupByA(f))
```
But to me it has at least two flaws:
1. It is too verbose and may require explicit types.
2. It mimics the idea of type safety for grouping: i.e. grouping over a regular collection ALWAYS results in zero or more groups of corresponding non-empty collections.

Reply:
to your point 2: putting `groupByA` on `NonEmptyTraverse` we could actually return `NonEmptyMap[B, F[A]]`, so we know both the map is non-empty and since `F[_]: NonEmptyTraverse` we also know `F[_]` is non-empty.

Reply:
Yes, this is a valid case too. But I'd suppose that grouping over a regular `Traverse` is a more common case anyway. And it's guaranteed to produce a regular `Map` of `NonEmpty*` groups. While grouping over `NonEmptyTraverse` is guaranteed to produce a `NonEmptyMap` of `NonEmpty*` groups. I mean both cases looks legit to me and I'd like to think on how to incorporate both of them.